megatron/checkpointing.py

```diff
@@ -736,3 +754,56 @@ def load_biencoder_checkpoint(model, only_query_model=False,
         print(' successfully loaded {}'.format(checkpoint_name))
 
     return model
+
+
+def _checkpoint_info():
+    args = get_args()
+    tokenizer = get_tokenizer()
+
+    return {
+        "padded_vocab_size": args.padded_vocab_size,
+        "original_vocab_size": tokenizer.vocab_size,
+    }
+
+def _universal_checkpoint_info():
```

Comment:
The VOCABULARY_PARAMETER_PATTERNS, PARAMETER_TO_AVERAGE_PATTERNS, PARAMETER_TO_AVERAGE_PATTERNS, PARAMETER_WITH_ROW_PARALLELISM_PATTERNS is model dependent.
Therefore, suggest the following:
- pass model as a parameter to this function (model is derived from MegatronModule)
- MegatronModule will implement "pure-virtual" methods to retrieve required information (e.g. will return a dict with above information)
- Each model (GPT3, LLaMA, Bloom) will have to implement the method(s) to retrieve their specific information

Reply:
Fair point. Since megatron code supports multiple models, it makes sense to push this logic to the model implementations.